torch/csrc/autograd/record_function_ops.cpp

```diff
@@ -28,23 +34,62 @@ at::Tensor record_function_enter(const std::string& name) {
   return at::cpp_custom_type_hack::create(std::move(rec), at::TensorOptions());
 }
 
+RecordFunction& getRecordFunctionFromTensor(const at::Tensor& handle) {
+  auto& rec = at::cpp_custom_type_hack::cast<RecordFunction>(handle);
+  return rec;
+}
+
 void record_function_exit(const at::Tensor& handle) {
   // We don't actually need to do anything with handle just need to persist the
   // lifetime until now.
-  auto& rec = at::cpp_custom_type_hack::cast<RecordFunction>(handle);
-  if (auto* current = rec.current()) {
-    AT_ASSERT(
-        current->name() == StringView("profiler::_record_function_exit"));
+  auto& rec = getRecordFunctionFromTensor(handle);
+  if (auto* current = RecordFunction::current()) {
+    AT_ASSERT(current->name() == StringView("profiler::_record_function_exit"));
```

Comment:
This `AT_ASSERT` exist prior to this PR. Shall replace it with TORCH_CHECK/TORCH_INTERNAL_ASSERT?